jfr-connection/src/main/java/io/opentelemetry/contrib/jfr/connection/Recording.java

```diff
@@ -0,0 +1,278 @@
+   *     of the recording.
+   * @param endTime The end time for the stream, or {@code null} to get data until the end of the
+   *     recording.
+   * @return An {@code InputStream}, or {@code null} if no data is available in the interval.
+   * @throws IOException A communication problem occurred when talking to the MBean server.
+   * @throws IllegalStateException If the {@code Recording} has not been stopped.
+   * @throws JfrStreamingException Wraps a {@code javax.management.JMException}.
+   * @see JfrStream#getDefaultBlockSize()
+   */
+  public InputStream getStream(Instant startTime, Instant endTime)
+      throws IOException, JfrStreamingException {
+    return getStream(startTime, endTime, JfrStream.getDefaultBlockSize());
+  }
+
+  /**
+   * Create a data stream for the specified interval using the given {@code blockSize}. The stream
+   * may contain some data outside the given range. The {@code blockSize} is used to configure the
+   * maximum number of bytes to read from underlying stream at a time. Setting blockSize to a very
+   * high value may result in an exception if the Java Virtual Machine (JVM) deems the value too
+   * large to handle. Refer to the javadoc for {@code
+   * jdk.management.jfr.FlightRecorderMXBean#openStream}.
+   *
+   * @param startTime The start time for the stream, or {@code null} to get data from the start time
+   *     of the recording.
+   * @param endTime The end time for the stream, or {@code null} to get data until the end of the
+   *     recording.
+   * @param blockSize The maximum number of bytes to read at a time.
+   * @return An {@code InputStream}, or {@code null} if no data is available in the interval.
+   * @throws IOException A communication problem occurred when talking to the MBean server.
+   * @throws IllegalStateException If the {@code Recording} has not been stopped.
+   * @throws JfrStreamingException Wraps a {@code javax.management.JMException}.
+   */
+  public InputStream getStream(Instant startTime, Instant endTime, long blockSize)
+      throws IOException, JfrStreamingException {
+    // state transitions: remain in state
+    State currentState = state.get();
+    if (currentState == State.STOPPED) {
+      return connection.getStream(id, startTime, endTime, blockSize);
+    } else {
+      throw new IllegalStateException(
+          createIllegalStateExceptionMessage(currentState, State.STOPPED));
+    }
+  }
+
+  /**
+   * Get the current state of this {@code Recording}.
+   *
+   * @return The current state of this {@code Recording}.
+   */
+  @Nullable
+  public State getState() {
+    return state.get();
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void close() throws IOException {
+    // state transitions:  any -> CLOSED
+    State oldState = state.getAndSet(State.CLOSED);
+    if (oldState == State.RECORDING) {
+      try {
+        connection.stopRecording(id);
+        connection.closeRecording(id);
+      } catch (Throwable ignored) {
```

Comment:
```suggestion
      } catch (Exception ignored) {
```

Generally not the best practice to catch `Error`s because they probably need to bubble up. Also, could it be confusing if `stopRecording()` throws and we don't attempt to `closeRecording()`?

Reply:
Done.